modules/siddhi-core/src/main/java/org/wso2/siddhi/core/query/processor/stream/window/BatchWindowProcessor.java

```diff
@@ -120,12 +141,51 @@ protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processo
                         expiredEventQueue.add(clonedStreamEvent);
                     } while (streamEventChunk.hasNext());
                 }
-                resetEvent = streamEventCloner.copyStreamEvent(streamEventChunk.getFirst());
+                streamEventChunkStore.add(streamEventChunk.getFirst());
```

Comment:
Why do you need to add this to a new chink. Can't we use the existing streamEventChunk